hudi-io/src/main/java/org/apache/hudi/common/util/StringUtils.java

```diff
@@ -369,6 +370,53 @@ public static String concatenateWithThreshold(String a, String b, int byteLength
     return a.substring(0, bestLength) + b;
   }
 
+  /**
+   * Concatenates the string representation of each object in the list
+   * and returns a single string. The result is capped at "lengthThreshold" characters.
+   *
+   * @param objectList      object list
+   * @param lengthThreshold number of characters to cap the string
+   * @param <T>             type of the object
+   *
+   * @return capped string representation of the object list
+   */
+  public static <T> String toStringWithThreshold(List<T> objectList, int lengthThreshold) {
```

Comment:
Could you please add UT as well?

Reply:
Will do.  This one is still in draft state.

Reply:
Fixed.